--- lua-xmlreader-0.1-old/Makefile	2008-01-28 14:42:14.000000000 -0200
+++ lua-xmlreader-0.1-new/Makefile	2008-01-28 14:44:16.000000000 -0200
@@ -4,11 +4,16 @@
 LUALIB= $(LUA)/lib
 LUABIN= $(LUA)/bin
 
+LIBXML2=/usr
+LIBXML2INC=$(LIBXML2)/include/libxml2
+LIBXML2LIB=$(LIBXML2)/lib
+
 # no need to change anything below here
 CFLAGS= $(INCS) $(DEFS) $(WARN) -O2
 WARN= -Wall
-INCS= -I$(LUAINC) -I/usr/include/libxml2/
-LIBS= -lxml2
+INCS= -I$(LUAINC) -I$(LIBXML2INC)
+LIBS= -lxml2 -L$(LIBXML2LIB)
+LIBFLAG= -shared
 
 OBJS= xmlreader.o error.o
 
@@ -17,7 +22,7 @@
 all: $(SOS)
 
 $(SOS): $(OBJS)
-	$(CC) -o $@ -shared $(OBJS) $(LIBS)
+	$(CC) -o $@ $(LIBFLAG) $(OBJS) $(LIBS)
 
 .PHONY: clean doc
 clean:
